src/hooks/useUsers.js

```diff
@@ -8,13 +8,23 @@ const { firebaseDb } = firebaseData;
 
 const useUsers = () => {
   const [filters, setFilters] = useState({ username: '' });
+  const [filtersChanged, setFiltersChanged] = useState(false);
   const [usersList, setUsersList] = useState();
 
   const { username } = filters;
 
+  const changeFilter = (key, value) => {
+    setFilters({
+      ...filters,
+      [key]: value,
+    });
```

Comment:
don't forget to use the previous value

```suggestion
    setFilters(prevState => ({
      ...prevState
      [key]: value,
    }));
```